File: thot-server/src/main/java/com/thot/server/model/domain/Template.java
package com.thot.server.model.domain;

import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Entity
@Getter
@Setter
public class Template {
    @Id
    private String id;

    private String name;

    @Column(columnDefinition = "TEXT")
    private String svg;

    @Column(columnDefinition = "TEXT")
    private String img;

    @Column(columnDefinition = "TEXT")
    private String xml;

    @ElementCollection
    private Set<String> markers;

}

File: thoth-web/src/app/common/types/template.ts
import {SecuredResource} from "./secured-resource";
import {Renderer} from "./renderer";

export interface Template extends SecuredResource {
  id: string;
  name: string;
  svg: string | undefined;
  img: string | undefined;
  xml: string | undefined;
  markers: string[];
  folder: string;
  usages: Renderer[]

}
